### ISSUE TITLE
Crash when X loses a display

### ISSUE THREAD

**Issue by pinballkeefer** [liballeg/allegro5#814]
When a display goes away (for example, gets unplugged), allegro crashes via the following route:

from _al_xwin_background_thread:
process_x11_event() [XEvent.type is 22]
[case ConfigureNotify] _al_xglx_display_configure_event()
_al_xglx_display_configure()
_al_xglx_get_monitor_info()

This calls xrandr_fetch_crtc which returns NULL, and then segfaults on crtc->x

Possibly a NULL check can be put in after that call and return false?  The bigger problem, I suppose, is the better handling of displays that go away?

I have a core dump I can refer back to if necessary.

**Comment by SiegeLord** [comment]
Is this windowed/fullscreen or fullscreen windowed?

**Comment by pinballkeefer** [comment]
It is definitely not windowed.  It is most likely fullscreen windowed, but it could possibly be fullscreen.  I will double check when I'm back at work tomorrow to be sure (we try both during display creation).

**Comment by pinballkeefer** [comment]
OK it gets created fullscreen, and then right after that gets changed (via al_set_display_flag) to fullscreen window.  I didn't write that part, but the comment is:  "Success - now make sure ALLEGRO_FULLSCREEN_WINDOW is on to get rid of unity top bar".  As far as I can tell stepping through, the call to al_set_display_flag is successful.

**Comment by SiegeLord** [comment]
I don't think we even have events for these things. One way to resolve this could be to send an event (like ALLEGRO_EVENT_MONITOR_LOST) and then mark the display as permanently lost (calling stuff on it shouldn't crash, but you should destroy it at the earliest convenience).

**Comment by SiegeLord** [comment]
A potential solution from Jesse Gordon from Discord:

```c
static bool xrandr_get_monitor_info(ALLEGRO_SYSTEM_XGLX *s, int adapter, ALLEGRO_MONITOR_INFO *mi)
{
   if(adapter < 0 || adapter >= (int)_al_vector_size(&s->xrandr_adaptermap))
      return false;

   int xscreen = _al_xglx_get_xscreen(s, adapter);
   xrandr_output *output = xrandr_map_adapter(s, xscreen, adapter);

   xrandr_crtc *crtc = xrandr_fetch_crtc(s, xscreen, output->crtc);

   if(crtc==0){return false;}  //Fixes segv when monitor shut off/turned on, by jesseg 7/31/2025

   mi->x1 = crtc->x;
   mi->y1 = crtc->y;
   mi->x2 = crtc->x + crtc->width; 
   mi->y2 = crtc->y + crtc->height;
   return true;
}
```
